### PR TITLE
Add missing headers

### PR DESCRIPTION
**Pull request by pastalian**
Some systems must explicitly include these headers.

Reported by: https://bugs.gentoo.org/922263

```
/var/tmp/portage/games-emulation/rpcs3-9999/work/rpcs3-9999/Utilities/Thread.cpp:3135:17: error: use of undeclared identifier 'SYS_gettid'
 3135 |         return syscall(SYS_gettid);
```
```
/var/tmp/portage/games-emulation/rpcs3-9999/work/rpcs3-9999/Utilities/JITASM.cpp:23:52: error: use of undeclared identifier 'getpid'; did you mean 'getpt'?
   23 |                 std::string name{fmt::format("/tmp/perf-%d.map", getpid())};
      |                                                                  ^~~~~~
      |                                                                  getpt
```
```
/var/tmp/portage/games-emulation/rpcs3-9999/work/rpcs3-9999/3rdparty/GL/../pine/pine_server.h:433:24: error: use of undeclared identifier 'read'; did you mean '_IO_cookie_io_functions_t::read'?
  433 |                                         auto tmp_length = read_portable(m_msgsock, &m_ipc_buffer[receive_length], MAX_IPC_SIZE - receive_length);
```

### CODE REVIEW

File: 3rdparty/pine/pine_server.h [hunk 1]
```diff
@@ -27,7 +27,7 @@
 #include <sys/un.h>
 #include <arpa/inet.h>
 #include <poll.h>
-#if defined(__DragonFly__) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__APPLE__)
+#if defined(__DragonFly__) || defined(__FreeBSD__) || defined(__linux__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__APPLE__)
```
Comment (MSuih) [inline 3rdparty/pine/pine_server.h:30]: You could just do `#ifndef MSVC` at this point

Comment (pastalian) [reply 3rdparty/pine/pine_server.h]: sure

Comment (pastalian) [reply 3rdparty/pine/pine_server.h]: Wait, it's right above this. Why was this even separated lol.

Comment (kd-11) [reply 3rdparty/pine/pine_server.h]: Since it's a platform check, _WIN32 would be the right define to check for.

**Pull request merged**